[BUG] Not possible to compile double battery GPIO control

**Describe the bug**
It is not possible to compile double battery with contactor control on Stark hardware

**To Reproduce**
Enable the following:
NISSAN_LEAF_BATTERY
DOUBLE_BATTERY
HW_STARK
CONTACTOR_CONTROL
CONTACTOR_CONTROL_DOUBLE_BATTERY

**Expected behavior**
Compilation succeeds

**Screenshots**

/home/dala/Git/Battery-Emulator/Software/src/communication/contactorcontrol/comm_contactorcontrol.cpp: In function 'void handle_contactors_battery2()':
/home/dala/Git/Battery-Emulator/Software/src/communication/contactorcontrol/comm_contactorcontrol.cpp:278:9: error: 'SECOND_BATTERY_CONTACTORS_PIN' was not declared in this scope
  278 |     set(SECOND_BATTERY_CONTACTORS_PIN, ON);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/dala/Git/Battery-Emulator/Software/src/communication/contactorcontrol/comm_contactorcontrol.cpp:281:9: error: 'SECOND_BATTERY_CONTACTORS_PIN' was not declared in this scope
  281 |     set(SECOND_BATTERY_CONTACTORS_PIN, OFF);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
exit status 1

Compilation error: 'SECOND_BATTERY_CONTACTORS_PIN' was not declared in this scope

**Version and settings (please complete the following information):**
- Please attach your USER_SETTINGS files for easier debugging
- Software version: 9.0.RC1
- Battery used: [NISSAN_LEAF]
- Inverter communication protocol: [ANY]
- Hardware used for Battery-Emulator: [HW_STARK]
- CONTACTOR_CONTROL: [yes]
- MQTT: [ANY]

--- a/Software/src/communication/contactorcontrol/comm_contactorcontrol.cpp
+++ b/Software/src/communication/contactorcontrol/comm_contactorcontrol.cpp
@@ -170,9 +170,9 @@ void handle_contactors() {
     handle_BMSpower();  // Some batteries need to be periodically power cycled
   }
 
-#ifdef CONTACTOR_CONTROL_DOUBLE_BATTERY
-  handle_contactors_battery2();
-#endif  // CONTACTOR_CONTROL_DOUBLE_BATTERY
+  if (contactor_control_enabled_double_battery) {
+    handle_contactors_battery2();
+  }
 
   if (contactor_control_enabled) {
     // First check if we have any active errors, incase we do, turn off the battery
@@ -272,17 +272,17 @@ void handle_contactors() {
   }
 }
 
-#ifdef CONTACTOR_CONTROL_DOUBLE_BATTERY
 void handle_contactors_battery2() {
+  auto second_contactors = esp32hal->SECOND_BATTERY_CONTACTORS_PIN();
+
   if ((contactorStatus == COMPLETED) && datalayer.system.status.battery2_allowed_contactor_closing) {
-    set(SECOND_BATTERY_CONTACTORS_PIN, ON);
+    set(second_contactors, ON);
     datalayer.system.status.contactors_battery2_engaged = true;
   } else {  // Closing contactors on secondary battery not allowed
-    set(SECOND_BATTERY_CONTACTORS_PIN, OFF);
+    set(second_contactors, OFF);
     datalayer.system.status.contactors_battery2_engaged = false;
   }
 }
-#endif
 
 /* PERIODIC_BMS_RESET - Once every 24 hours we remove power from the BMS_power pin for 30 seconds.
 REMOTE_BMS_RESET - Allows the user to remotely powercycle the BMS by sending a command to the emulator via MQTT.

